빠른 대전 전적 가져오는 로직 수정

yokxim2 opened the pull request:
## ✨ 담당 파트
- 빠른 대전 로직 수정

## 🔎 작업 상세 내용
- 빠른 대전 데이터를 랭크 게임 데이터를 쌓는 것처럼 수정했습니다.

** 상세 목록
승률 계산 로직 획일화: 모든 경기 수 체크 후 승 / 판수 로 계산
KDA 계산 획일화: 모든 경기 체크 후 K+A / D 로 계산
최근 선호 챔피언 획일화: 모든 경기 체크 후 Favorite 데이터로 저장

## 🔧 앞으로의 과제
- 카프카를 사용해서 프로듀서 & 컨슈머 구현하기

## ✅ 테스트 코드 작성 및 기능 테스트 여부
- [] 테스트 코드 작성
- [] 기능 테스트 여부

## ➕ 이슈 링크
- #35

yokxim2 committed:
fix(riot-client-service) 일반 게임 로직 수정 #35

- 최근 20게임만 저장하던 빠른 대전 전적을, 시즌 내 모든 데이터를 수집하는 로직으로 수정했습니다.

yokxim2 committed:
fix(riot-client-service) 일반 게임 로직 수정 #35

- 최근 20게임만 저장하던 빠른 대전 전적을, 시즌 내 모든 데이터를 수집하는 로직으로 수정했습니다.

yokxim2 committed:
Revert "fix(riot-client-service) 주석 추가 및 작은 리팩토링 #32"

This reverts commit 13a0c11ee9e9c58e649bb4f801f8716c42140ab8.

yokxim2 committed:
fix(account-game-data-service) 매치 정보 수정/생성 시 플레이한 전적이 없을 때 상황 처리 #35

yokxim2 committed:
fix(account-game-data-service) 매치 정보 수정/생성 시 플레이한 전적이 없을 때 상황 처리 추가 로직 #35

- 일반 게임 역시 update 요청 시 새 객체를 만들어서 집어 넣는 로직이 아닌, 기존에 있던 데이터를 update하는 로직을 추가했습니다.

yokxim2 committed:
feat(quick-game-data) update 로직 추가 #35

yokxim2 committed:
chore(riot-client-service) 주석 추가 및 자잘한 로직 변경 #35

- += -> =

yokxim2 committed:
delete(riot-data-provider) delete useless class #35

yokxim2 committed:
fix(riot-data-service) unlink meaningless provider #35

for2ver0 reviewed (approved):
변경 사항 모두 확인했습니다 👍 👍 👍 
수고하셨습니다!!!

yokxim2 committed:
feat(favorite) Favorite 기존 값 업데이트 메서드 생성 #35

yokxim2 committed:
feat(favorite-repository) accountId, queueType, championName 값으로 조회 #35

yokxim2 committed:
fix(riot-client-service) getMatchStats: Favorite 정보 업데이트 로직 추가 #35

- 바뀌기 전 로직은 matchIds로 들어오면 해당 accountId, queueType으로 Favorite DB를 전체 초기화하고 다시 입력된 매치 정보로 값을 다시 할당하는 방법이었습니다. 이 로직은 오직 AccountGameData 엔티티가 생성될 때 사용했던 로직이었습니다.
- 하지만 이제는 AccountGameData가 update될 수 있기 때문에, 계정의 모든 매치 데이터가 getMatchStats에 들어가지 않을 수가 있습니다. 따라서 기존에 존재하던 Favorite 값에 신규 조회 내용을 반영해 주어야 할 필요성이 생겼습니다.

LEEJI-HOON1 reviewed (approved):
수고하셨습니다.
